app/policies/article_policy.rb

```diff
@@ -144,13 +145,42 @@ def subscriptions?
     user_author? || user_super_admin?
   end
 
+  def elevated_user?
+    user_any_admin? || user_moderator?
+  end
+
   # this method performs the same checks that determine
-  # if the record can be featured
+  # if the record can be featured or if user can adjust
+  # any tag
   def revoke_publication?
     require_user!
     return false unless @record.published?
 
-    user_any_admin? || user_moderator?
+    elevated_user?
+  end
+
+  def allow_tag_adjustment?
+    require_user!
+
+    elevated_user? || tag_moderator_eligible?
+  end
+
+  def tag_moderator_eligible?
+    tag_ids_moderated_by_user = Tag.with_role(:tag_moderator, @user).ids
+    return false unless tag_ids_moderated_by_user.size.positive?
+
+    adjustments = TagAdjustment.where(article_id: @record.id)
+    has_room_for_tags = @record.tag_list.size < MAX_TAG_LIST_SIZE
+    # ensures that mods cannot adjust an already-adjusted tag
+    has_no_relevant_adjustments = adjustments.pluck(:tag_id).intersection(tag_ids_moderated_by_user).size.zero?
```

Comment:
not sure how many `tag_ids` there could potentially be in the intersection of `adjustments` and `tag_ids_moderated_by_user`, but, it _might_ be worth considering using `empty?` in favor of counting then checking if the array is `zero?`

Reply:
Thank you @dscottS3